server/src-lib/Hasura/Server/Auth/JWT.hs

```diff
@@ -389,23 +419,42 @@ instance J.FromJSON JWTConfig where
   parseJSON = J.withObject "JWTConfig" $ \o -> do
     mRawKey <- o J..:? "key"
     claimNs <- o J..:? "claims_namespace"
+    claimNsPathStr <- o J..:? "claims_namespace_path"
     aud     <- o J..:? "audience"
     iss     <- o J..:? "issuer"
     jwkUrl  <- o J..:? "jwk_url"
     isStrngfd <- o J..:? "claims_format"
 
-    case (mRawKey, jwkUrl) of
-      (Nothing, Nothing) -> fail "key and jwk_url both cannot be empty"
-      (Just _, Just _)   -> fail "key, jwk_url both cannot be present"
-      (Just rawKey, Nothing) -> do
-        keyType <- o J..: "type"
-        key <- parseKey rawKey keyType
-        return $ JWTConfig (Left key) claimNs aud isStrngfd iss
-      (Nothing, Just url) ->
-        return $ JWTConfig (Right url) claimNs aud isStrngfd iss
+    let hasuraClaimsNs :: Either String JWTConfigClaims
```

Comment:
You can directly `parseJSONPath` and `fail` in single line
```haskell
     maybeClaimNsPath <- (o J..:? "claims_namespace_path")  >>= mapM (either failJSONPathParsing pure . JSONPath.parseJSONPath)
```
in `where`
```haskell
failJSONPathParsing err = fail $ "invalid JSON path claims_namespace_path error: " ++ err
```
Compute `hasuraClaimsNs` without any error handling

```haskell
hasuraClaimsNs = case maybeClaimNsPath of
  Just claimsNsPath, -> pure $ ClaimNsPath claimsNsPath
  Nothing -> pure $ ClaimNs $ fromMaybe defaultClaimNs claimNs
```

Now code is much read-able and pretty. :slightly_smiling_face: